### PR TITLE
fix: stop inheriting superchain interfaces

### PR DESCRIPTION
**Pull request by agusduha**

### CODE REVIEW

File: packages/contracts-bedrock/src/L2/OptimismSuperchainERC20Factory.sol [hunk 1]
```diff
@@ -30,6 +25,10 @@ contract OptimismSuperchainERC20Factory is IOptimismSuperchainERC20Factory, ISem
     /// @custom:semver 1.0.0-beta.3
     string public constant version = "1.0.0-beta.3";
 
+    /// @notice Mapping of the deployed OptimismSuperchainERC20 to the remote token address.
+    ///         This is used to keep track of the token deployments.
+    mapping(address _localToken => address remoteToken_) public deployments;
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/src/L2/OptimismSuperchainERC20Factory.sol:30]:
```suggestion
    mapping(address _localToken => address _remoteToken) public deployments;
```

File: packages/contracts-bedrock/test/L2/SuperchainERC20Bridge.t.sol [hunk 2]
```diff
@@ -62,9 +62,18 @@ contract SuperchainERC20BridgeTest is Bridge_Initializer {
...
+        address _to,
+        uint256 _amount,
+        uint256 _chainId,
+        bytes32 _msgHash
+    )
+        external
+    {
+        // Ensure `_sender` and `_to` is not the zero address
         vm.assume(_sender != ZERO_ADDRESS);
+        vm.assume(_to != ZERO_ADDRESS);
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainERC20Bridge.t.sol:76]: Weird that it worked w/o this assume

**Review by 0xDiscotech — APPROVED** [review]

File: packages/contracts-bedrock/test/L2/SuperchainERC20Bridge.t.sol (hunk 2 above)
Comment (agusduha) [reply packages/contracts-bedrock/test/L2/SuperchainERC20Bridge.t.sol:76]: It was a later change

File: packages/contracts-bedrock/src/L2/OptimismSuperchainERC20Factory.sol (hunk 1 above)
Comment (agusduha) [reply packages/contracts-bedrock/src/L2/OptimismSuperchainERC20Factory.sol:30]: Now returned variables must end with `_`, all interfaces have changed

File: packages/contracts-bedrock/src/L2/SuperchainERC20.sol [hunk 3]
```diff
@@ -19,7 +22,7 @@ abstract contract SuperchainERC20 is ERC20, ISuperchainERC20, ISemver {
 
     /// @notice Semantic version.
     /// @custom:semver 1.0.0-beta.1
-    function version() external pure virtual returns (string memory) {
+    function version() external view virtual returns (string memory) {
```
Comment (gotzenx) [inline packages/contracts-bedrock/src/L2/SuperchainERC20.sol:25]: why not pure?

Comment (agusduha) [reply packages/contracts-bedrock/src/L2/SuperchainERC20.sol:25]: ABI diff makes the script fail

**Review by gotzenx — APPROVED** [review]

**Pull request merged**